apps/io-services-cms-webapp/src/utils/subscription.ts

```diff
@@ -0,0 +1,64 @@
+import { ApiManagementClient } from "@azure/arm-apimanagement";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseErrorNotFound,
+  IResponseErrorTooManyRequests,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+} from "@pagopa/ts-commons/lib/responses";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as TE from "fp-ts/lib/TaskEither";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { IConfig } from "../config";
+import { getSubscription } from "../lib/clients/apim-client";
+
+export type ErrorResponses =
+  | IResponseErrorNotFound
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+  | IResponseErrorTooManyRequests;
+
+/**
+ * Using the **API Manage key** as 'Ocp-Apim-Subscription-Key', the Subscription relating to this key will have a name starting with "MANAGE-"
+ * and accordingly no longer equal to the serviceId.
+ *
+ * Therefore, since it is no longer possible to verify the equality *subscriptionId == serviceId*,
+ * it is necessary to verify that the owner of the subscription of the API Key is the same owner of the Subscription to which the ServiceId belongs
+ *
+ * @param config
+ * @param apimClient
+ * @param serviceId
+ * @param ownerSubscriptionId subscriptionId related to 'Ocp-Apim-Subscription-Key'
+ * @param userId APIM userId
+ * @returns
+ */
+export const serviceOwnerCheckManageTask = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  serviceId: NonEmptyString,
+  ownerSubscriptionId: NonEmptyString,
+  userId: NonEmptyString
+): TaskEither<ErrorResponses, NonEmptyString> =>
+  pipe(
+    getSubscription(
+      apimClient,
+      config.AZURE_APIM_RESOURCE_GROUP,
+      config.AZURE_APIM,
+      serviceId
+    ),
+    TE.mapLeft(
+      () =>
+        ResponseErrorInternal(
+          `An error has occurred while retrieving service '${serviceId}'`
+        ) // FIXME: handle the error correctly
+    ),
+    TE.chainW((serviceSubscription) =>
+      ownerSubscriptionId.startsWith("MANAGE-")
```

Comment:
If I understand well when `ownerSubscriptionId.startsWith("MANAGE-") === false` we don't need to call `getSubscription` API (which we do anyway), isn't it? You probably want to handle this case first. I'd give a name to this boolean check extracting it in its own method:

`const isManageKey = (ownerSubscriptionId: NonEmptyString) => ownerSubscriptionId.startsWith("MANAGE-");
`

Reply:
Thanks for the suggested improvement, we have made the change and now we execute as the first thing in the pipe that the passed key is a manageKey.

``` typescript
...
  pipe(
    ownerSubscriptionId,
    TE.fromPredicate(
      (oSId: NonEmptyString) => isManageKey(oSId),
      () => ResponseErrorForbiddenNotAuthorized
    ),
    TE.chainW(() =>
      pipe(
        getSubscription(
          apimClient,
...
```